### PR TITLE
Add missing preview features of Deployment and Deployment Statuses API

### PR DESCRIPTION
**Pull request by pascal-hofmann**
- Enable support for deployment status states `in_progress` and `queued`
  - Add arguments `production_environment` and `transient_environment` to `Repository.create_deployment()`
  - Add attributes `production_environment` and `transient_environment` to `Deployment`
  - Fix incorrect assertion for `description` in `Deployment.create_status()`
  - Add arguments `environment`, `environment_url`, `auto_inactive` to` Deployment.create_status()`
  - Add attribute `environment_url` to `DeploymentStatus`

References:
  - https://docs.github.com/en/rest/reference/repos#deployments
  - https://developer.github.com/changes/2016-04-06-deployment-and-deployment-status-enhancements/
  - https://developer.github.com/changes/2018-10-16-deployments-environments-states-and-auto-inactive-updates/

### CODE REVIEW

File: github/Deployment.py
Comment (s-t-e-v-e-n-k) [inline github/Deployment.py:91]: This is an API break.

File: github/ApplicationOAuth.pyi
Comment (s-t-e-v-e-n-k) [inline github/ApplicationOAuth.pyi:15]: Why are you making these changes?

File: github/Branch.py
Comment (s-t-e-v-e-n-k) [inline github/Branch.py:166]: As above -- why are you making these changes?

**Review by s-t-e-v-e-n-k — CHANGES_REQUESTED** [review]
There is a awful lot more here changed than what the PR says, would you mind cleaning this up to just doing that to make this easier to review?

File: github/ApplicationOAuth.pyi
Comment (pascal-hofmann) [reply github/ApplicationOAuth.pyi:15]: Tox made these. I just ran it.

File: github/Branch.py
Comment (pascal-hofmann) [reply github/Branch.py:166]: Tox made these. I just ran it. I'll amend my commit and not include these changes.

File: github/Deployment.py
Comment (pascal-hofmann) [reply github/Deployment.py:91]: If I remember correctly the payload returned already was a string, even if stated otherwise here. I'll have another look at this.

**Comment by pascal-hofmann** [comment]
@s-t-e-v-e-n-k I've removed the linted files. Now the build fails again: https://github.com/PyGithub/PyGithub/pull/1674/checks?check_run_id=1049885210

I guess this should be fixed on master?

File: github/Deployment.py
Comment (pascal-hofmann) [reply github/Deployment.py]: Github just returns whatever is passed. No matter which type. I've changed the assert in create_deployment to dict and reverted my changes.

**Comment by s-t-e-v-e-n-k** [comment]
```
lint run-test: commands[0] | pre-commit run --all-files --show-diff-on-failure
black....................................................................Passed
```

is what I get locally, so I'm still digging into why CI has no love.

**Comment by pascal-hofmann** [comment]
@s-t-e-v-e-n-k I think the CI issue is independent from my PR (other PRs have the same problem). Can this be merged or are there still things I should improve?

**Comment by s-t-e-v-e-n-k** [comment]
I haven't had time yet to dig into the failure yet, I hope to get some time tomorrow.

**Comment by pascal-hofmann** [comment]
@s-t-e-v-e-n-k I rebased this on latest master. Now all the checks pass.

**Comment by pascal-hofmann** [comment]
@s-t-e-v-e-n-k It would be great to have this merged and packed in a new release of PyGit.

Is there anything I can do to help with this?

**Comment by pascal-hofmann** [comment]
I just noticed that the commit message was outdated. This is now fixed.

**Comment by pascal-hofmann** [comment]
@adamtheturtle @jzelinskie @nhomar @jacquev6 @sfdye @s-t-e-v-e-n-k 

It would be great to get this merged and released. I noticed that there is not much of maintainer activity lately and quite some outstanding contributions. I volunteer as maintainer. This is a great project, I want to move it forward.

**Comment by sfdye** [comment]
@pascal-hofmann Thanks for your proposal, let's talk in private.

File: github/Deployment.py
Comment (s-t-e-v-e-n-k) [inline github/Deployment.py:170]: This is repeated a fair amount -- would you consider a static method in this class that returns the string?

Comment (pascal-hofmann) [reply github/Deployment.py]: This is now implemented.

File: github/Repository.py
Comment (s-t-e-v-e-n-k) [inline github/Repository.py:1852]: This is no longer json

Comment (s-t-e-v-e-n-k) [reply github/Repository.py:1852]: Further more, we need to remember to mention the API break on the next release.

**Review by s-t-e-v-e-n-k — CHANGES_REQUESTED** [review]
This is looking so close, thanks for your changes. Some doc changes, and you're missing typing hint changes to Repository.pyi and I think we can finally merge this!

**Comment by pascal-hofmann** [comment]
Hi @s-t-e-v-e-n-k,
sorry for the delay. I updated the stub files, fixed the payload type and rebased on master. I also moved code for `production_environment()` and `transient_environment()` to a place within `github/Deployment.py` where it makes more sense.

I did a separate commit with these changes for now, so you can easily see latest changes. Give me a shout if I should squash the commits.

**Comment by s-t-e-v-e-n-k** [comment]
Apparently, we have the technology to squash commits at merge time. :-)

File: github/Repository.py
Comment (pascal-hofmann) [reply github/Repository.py]: This is now implemented.

**Review by s-t-e-v-e-n-k — APPROVED** [review]

File: github/Repository.pyi
Comment (s-t-e-v-e-n-k) [inline github/Repository.pyi:311]: I mean, surely this is Deployment, not Any. However, I don't want to hold this PR any further.

Comment (pascal-hofmann) [reply github/Repository.pyi:311]: Good catch. Fixed in  https://github.com/PyGithub/PyGithub/pull/1767/

**Comment by pascal-hofmann** [comment]
@s-t-e-v-e-n-k Thanks for all your work on this PR and this project in general!

**Pull request merged**
